R/use_lintr.R

```diff
@@ -0,0 +1,45 @@
+#' Use lintr in your project
+#'
+#' Create a minimal lintr config file as a starting point for customization
+#'
+#' @param path Path to project root, where a `.lintr` file should be created.
+#' If it already exists, an error will be thrown.
+#' @param type What kind of configuration to create?
+#'
+#'  * `minimal` creates a minimal lintr config, based off the default linters ([linters_with_defaults()]).
+#'  * `full` creates a lintr config using all available linters using [linters_with_tags()].
+#'
+#' @return Path to the generated configuration, invisibly.
+#'
+#' @export
+#' @examples
+#' \dontrun{
+#' # use the default set of linters
+#' lintr::use_lintr()
+#' # or try all linters
+#' lintr::use_lintr(type = "full")
+#'
+#' # then
+#' lintr::lint_dir()
+#' }
+use_lintr <- function(path = ".", type = c("minimal", "full")) {
+  config_file <- normalizePath(file.path(path, getOption("lintr.linter_file")), mustWork = FALSE)
+  if (file.exists(config_file)) {
+    stop("File '", config_file, "' already exists.")
```

Comment:
Maybe not the most helpful error, since we are generating that path from the user's input.

`"Found an existing configuration file at '", config_file, "'"`?